src/docs/development/platform-integration/platform-views.md

```diff
@@ -196,58 +243,299 @@ public class MainActivity extends FlutterActivity {
+  [[self registrarForPlugin:@"<plugin-name>"] registerViewFactory:factory
+                                                          withId:@"<platform-view-type>"];
+  return [super application:application didFinishLaunchingWithOptions:launchOptions];
+}
+
+@end
+```
+
+For plugin registration, modify the main plugin file (e.g. `FLPlugin.m`):
+
+```objc
+#import "FLPlugin.h"
+#import "FLNativeView.h"
+
+@implementation FLPlugin
+
++ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
+  FLNativeViewFactory* factory =
+      [[FLNativeViewFactory alloc] initWithMessenger:registrar.messenger];
+  [registrar registerViewFactory:factory withId:@"<platform-view-type>"];
+}
+
+@end
+```
+
+For more information, see the API docs for:
+
+* [`FlutterPlatformViewFactory`][]
+* [`FlutterPlatformView`][]
+* [`PlatformView`][]
+
+[`FlutterPlatformViewFactory`]: {{site.api}}/objcdoc/Protocols/FlutterPlatformViewFactory.html
+[`FlutterPlatformView`]: {{site.api}}/objcdoc/Protocols/FlutterPlatformView.html
+[`FlutterPlugin`]: {{site.api}}/objcdoc/Protocols/FlutterPlugin.html
 [`UIKitView`]: {{site.api}}/flutter/widgets/UiKitView-class.html
+
+## Putting it together
+
+When implementing the `build()` method in Dart, you can use [`defaultTargetPlatform`][]
+to detect the platform, and decide what widget to use:
+
+<!-- skip -->
+```dart
+Widget build(BuildContext context) {
+  // This is used in the platform side to register the view.
+  final String viewType = '<platform-view-type>';
+  // Pass parameters to the platform side.
+  final Map<String, dynamic> creationParams = <String, dynamic>{};
+
+  switch (defaultTargetPlatform) {
+    case TargetPlatform.android:
+      // return widget on Android.
+    case TargetPlatform.iOS:
+      // return widget on iOS.
+    default:
+      throw UnsupportedError("Unsupported platform view");
+  }
+}
+```
+
+[`defaultTargetPlatform`]: {{site.api}}/flutter/foundation/defaultTargetPlatform.html
+
+
+### Performance
+
+Platform views in Flutter come with performance trade-offs.
+
+For example, in a typical Flutter app, the Flutter UI is composed
+on a dedicated raster thread. This allows Flutter apps to be fast,
+as the main platform thread is rarely blocked.
+
+While a platform view is rendered, the Flutter UI is composed from
+the platform thread, which competes with other tasks like
+handling OS or plugin messages, etc.
+
+Prior to Android 10, Hybrid composition copies each Flutter frame
+out of the graphic memory into main memory and then copied back to
+a GPU texture. As this copy happens per frame, the performance of
+the entire Flutter UI may be impacted.
+
+On the other hand, Virtual display makes each pixel of the native view
+flow through additional intermediate graphic buffers, which cost graphic
+memory and drawing perfomance.
+
+For complex cases, there are some techniques that can be used to mitigate
+these issues.
+
```

Comment:
Missing section on the performance impact of hybrid composition on Android >= 10

Reply:
Done.